File: Flowspire-app/src/app/models/RegisterRequest.ts
export interface RegisterRequest {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  phoneNumber: string;
  birthDate?: Date | string;
  gender?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  country?: string;
  postalCode?: string;
  role?: string;
}

File: Flowspire-api/Flowspire.API/Models/RegisterRequest.cs
using Flowspire.Domain.Enums;
using System;

namespace Flowspire.API.Models
{
    public class RegisterRequest
    {
        public string Email { get; set; }
        public string NameFirst { get; set; }
        public string LastName { get; set; }
        public string Password { get; set; }
        public string PhoneNumber { get; set; }
        public DateTime? BirthDate { get; set; }
        public Gender Gender { get; set; } = Gender.NotSpecified;
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? Country { get; set; }
        public string? PostalCode { get; set; }
        public UserRole Role { get; set; }
    }
}
